packages/opencensus-core/src/stats/model/metrics/counter.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as logger from '../../../common/console-logger';
+import {Logger} from '../../../common/types';
+import {Measure, MeasureUnit, Tags} from '../types';
+
+import {BaseMetric} from './metric';
+import {MetricConfig, MetricDescriptor, MetricSingleValues, MetricValues, MetricValuesTypes} from './types';
+import {Counter} from './types';
+
+
+/**
+ * The counter is a record mechanism that counts how many times an event occurs.
+ * It cannot be decremented.
+ *
+ * NumberCounter is a Counter Recorder Mechanism class using number
+ */
+export class NumberCounter implements Counter<number> {
+  private logger: Logger;
+  private localValue = 0;
+  timestamp: number;
+  startTime: number;
+  endTime: number;
+
+  constructor(alogger: Logger) {
+    this.logger = alogger || logger.logger();
+    this.startTime = Date.now();
```

Comment:
The constructor doesn't initialize `endTime` or `timestamp`. Maybe call `reset()` here?